### PR TITLE
Display type when hovering over function arguments

### LINKED ISSUE #2240: LSP: Display type when hovering over function parameter
Consider the following as an example:
```gleam
fn first_if_true(test, first, second) {
  case test(first) {
    True -> first
    False -> second
  } 
}
```
When hovering over `test`, `first`, or `second` inside the function head, the type information is not displayed. 
You would need to hover over instances of those parameters inside the function body to get the type information.

### PR DESCRIPTION
**Pull request by Chriscbr**
Initial pass at adding LSP hover support for function arguments.

Closes #2240

### CODE REVIEW

File: compiler-core/src/build.rs [hunk 1]
```diff
@@ -266,6 +267,7 @@ pub enum Located<'a> {
     Statement(&'a TypedStatement),
     Expression(&'a TypedExpr),
     ModuleStatement(&'a TypedDefinition),
+    FunctionArgument(&'a TypedArg),
```
Comment (Chriscbr) [inline compiler-core/src/build.rs:270]: I wasn't too sure if adding a new enum case here would be appropriate or not.

**Comment by nino** [comment]
I think this functionality is included in the [PR I made a while ago](https://github.com/gleam-lang/gleam/pull/2302) (assuming that will be merged at some point). (Sorry for not tagging the function argument ticket in the PR!)

**Comment by lpil** [comment]
Very sorry @Chriscbr , but I have just merged @nino's patch. Thank you for your contribution though, the code looked good!

**Comment by Chriscbr** [comment]
My bad! Thanks for the heads up 👍

**Pull request closed without merging**